spec/requests/legalaid_application_spec.rb

```diff
@@ -59,3 +50,44 @@
     end
   end
 end
+
+RSpec.describe 'create Legal aid applications with or without proceedings' do
+  let(:response_json) { JSON.parse(response.body) }
+
+  describe 'POST /v1/applications' do
+    it 'returns an application reference' do
+      post '/v1/applications'
+      expect(response_json['data']['attributes']['application_ref']).not_to be_empty
+      expect(response.status).to eql(201)
+      expect(response.content_type).to eql('application/json')
+    end
+
+    context 'when proceedings are saved along with application reference' do
+      let(:proceeding_type_first) { ProceedingType.create(code: 'PR0001', ccms_code: 'PBM23', meaning: 'test meaning1', description: 'test desc1') }
+      let(:proceeding_type_second) { ProceedingType.create(code: 'PR0002', ccms_code: 'PBM24', meaning: 'test meaning2', description: 'test desc2') }
+
+      it 'creates a new legal aid application without proceeding_type_codes ' do
+        expect do
+          post '/v1/applications'
+        end.to change { LegalAidApplication.count }.by(1)
+      end
+
+      it 'creates a new legal aid application with proceeding_type_codes' do
+        post '/v1/applications', params: { proceeding_type_codes: [proceeding_type_first.code] }
+        application_ref = response_json['data']['attributes']['application_ref']
+        expect(LegalAidApplication.find_by(application_ref: application_ref).proceeding_types.size).to eq(1)
+      end
+
+      it 'creates a new legal aid application with more than one proceeding_type_codes' do
+        post '/v1/applications', params: { proceeding_type_codes: [proceeding_type_first.code, proceeding_type_second.code] }
+        application_ref = response_json['data']['attributes']['application_ref']
+        expect(LegalAidApplication.find_by(application_ref: application_ref).proceeding_types.size).to eq(2)
+      end
+
+      it 'creates a new legal aid application when proceeding_type_codes param is not valid, response should be bad data ' do
+        post '/v1/applications', params: { proceeding_type_codes: ['invalid_proceeding_type_code'] }
+        expect(response.status).to eql(400)
```

Comment:
Also, for error cases we should probably be asserting on the format of the error JSON payload returned to the client.

Reply:
done